op-service/sources/eth_client.go

```diff
@@ -352,26 +315,31 @@ func (s *EthClient) PayloadByLabel(ctx context.Context, label eth.BlockLabel) (*
 // It verifies the receipt hash in the block header against the receipt hash of the fetched receipts
 // to ensure that the execution engine did not fail to return any receipts.
 func (s *EthClient) FetchReceipts(ctx context.Context, blockHash common.Hash) (eth.BlockInfo, types.Receipts, error) {
+	receipts, ok := s.recProvider.CachedReceipts(blockHash)
+	if ok {
+		// should always hit the cash
+		info, err := s.InfoByHash(ctx, blockHash)
+		if err != nil {
+			return nil, nil, fmt.Errorf("querying header: %w", err)
```

Comment:
Style question, do we want the message to be `error querying header` or is that redundant, as the message would be coming in attached to an `err`? (Is that something even defined by our style guidelines?)

Reply:
Right now, we don't have a style guide on how to format error messages and it's very inconsistent. I personally like to follow the patterns that I see in other big projects, like golang itself, that make sense to me. E.g. they also use the `doing foo: %w` style when wrapping errors sometimes, like [here](https://github.com/golang/go/blob/abf84221fc35a03848cda60373adf484eafb3e1b/src/cmd/go/internal/imports/scan.go#L61).

The reason why I think this makes sense is if you look at how an error chain will look like when printed in full. If an error bubbles up the stack somewhere, there's e.g. a `log.Errorf("Error in main loop: %v", err)`.
* using my pattern, we get something like `Error in main loop: doing x: opening y: file not found`
* repeating that something failed would look like `Error in main loop: failed to do x: failed to open y: file not found` which is more noisy imo.

So imo in general the only place where one should use the word "error" or "failed" should be high up the stack when the error chain is printed. Intermediate messages in an error chain should just contain enough context to understand the error trace. Every time adding "failed to do y:" or "error doing x:" looks noisy to me and doesn't add more context. But that's really all just my personal preference right now 😄